src/pages/GamePage.jsx

```diff
@@ -8,72 +8,57 @@ const ranArr = Array.from({ length: numbersQuantity }, () =>
   Math.floor(Math.random() * numbersRange)
 );
 const triesLimit = 5;
-const selectedArray = [];
-let filteredArray = [];
 
 const GamePage = () => {
   const [userNumber, setUserNumber] = useState(false),
     [errorHandler, setErrorHandler] = useState(false),
     [resultHandler, setResultHandler] = useState(false),
     [repeatError, setRepeatError] = useState(false),
     [triesCounter, setTriesCounter] = useState(0);
-
+  let resVal = false;
+  let checkVal = false;
+  let evenCheck = false;
   const inputHandler = (text) => {
     setUserNumber(parseInt(text.target.value));
   };
   const saveUserNumber = (event) => {
-    let checkVal = false;
+    if (ranArr[0] % 2 === 0) {
+      evenCheck = true;
+    }
     setTriesCounter(triesCounter + 1);
     if (isNaN(userNumber)) {
       setErrorHandler("Podaj cyfre");
     } else {
       event.preventDefault();
       setErrorHandler(false);
-      filteredArray = evenFilter(ranArr);
-      selectedArray.forEach((elem) => {
-        if (elem === userNumber) {
-          checkVal = true;
-        } else {
-          checkVal = false;
-        }
-      });
+      ranArr.forEach((elem, index, arr) =>
+        checkArray(elem, index, arr, userNumber, numbersQuantity)
```

Comment:
👍

Reply:
dobrze z funkcja, ale co to jest `numbersQuantity`?
czemu przekazujesz tablice?